Use literals to create lists and dicts

adamchainz opened the pull request:
They’re clearer and a little bit faster, per [rule C408 in flake8-comprehensions](https://github.com/adamchainz/flake8-comprehensions?tab=readme-ov-file#c408-unnecessary-dictlisttuple-call---rewrite-as-a-literal) (which I’d be happy to add to the setup, if you want).

matthewhegarty commented:
looks good, thanks.  Feel free to add the rule so we don't get a mixture of styles in future.

adamchainz commented:
I added flake8-comprehensions and fixed a few other issues it found:

```
import_export/forms.py:133:21: C416 Unnecessary list comprehension - rewrite using list().
import_export/forms.py:265:27: C416 Unnecessary list comprehension - rewrite using list().
import_export/forms.py:267:16: C419 Unnecessary list comprehension passed to any() prevents short-circuiting - rewrite as a generator.
tests/core/tests/test_resources/test_modelresource/test_export.py:82:26: C408 Unnecessary dict call - rewrite as a literal.
```

matthewhegarty reviewed (approved):
Thanks
